File: L1863-SumOfAllSubsetXorTotals.java
// https://leetcode.com/problems/sum-of-all-subset-xor-totals/
class Solution {
    public static void dfs(int i, int[] nums, int length, List<Integer> res, List<Integer> subset) {
        if (i >= length) {
            int r = 0;
            for (int num: subset) {
                r ^= num;
            }
            res.add(r);
            return;
        }
        subset.add(nums[i]);
        dfs(i + 1, nums, length, res, subset);
        
        subset.remove(subset.size() - 1);
        dfs(i + 1, nums, length, res, subset);
    }
    public int subsetXORSum(int[] nums) {
        List<Integer> res = new ArrayList<>();
        dfs(0, nums, nums.length, res, new ArrayList<>());
        int result = 0;
        for (int num: res) {
            result += num;
        }
        return result;
    }
}

File: L1863-sum_of_all_subset_xor_totals.py
# https://leetcode.com/problems/sum-of-all-subset-xor-totals/
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        res = []  
        subset = []
        def dfs(i):
            if i >= len(nums):
                r = 0
                for i in subset:
                    r ^= i
                res.append(r)
                return
            # decision to include nums[i]
            subset.append(nums[i])
            dfs(i + 1)
            
            # decision not to include nums[i]
            subset.pop()
            dfs(i + 1)
            
        dfs(0)
        resSum = 0
        for num in res:
            resSum += num
        return resSum
